konnectivity-client/go.mod

```diff
@@ -7,7 +7,7 @@ go 1.17
 require (
 	github.com/prometheus/client_golang v1.11.1
 	go.uber.org/goleak v1.2.0
-	golang.org/x/net v0.7.0 // indirect
+	golang.org/x/net v0.17.0 // indirect
```

Comment:
client lib considerations:

Konnectivity `release-0.1` branch is used by k/k minor versions 1.26 and 1.27. I verified that https://github.com/kubernetes/kubernetes/pull/121117 has already been cherry picked to the relevant release branches there.